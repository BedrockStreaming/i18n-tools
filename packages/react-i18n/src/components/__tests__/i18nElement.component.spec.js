import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import { HtmlTrans } from '../i18nElement.component';

describe('i18n.renderProps', () => {
  let context = {};
  const childContextTypes = {
    getTranslateFunction: PropTypes.func,
  };

  beforeEach(() => {
    context = {
      getTranslateFunction: jest.fn(() => jest.fn(x => `<div>${x}</div>`)),
    };
  });

  const getWrapper = (props) => shallow(<HtmlTrans i18nKey="foo.bar" {...props} />, { context, childContextTypes });

  it('should return html translation', () => {
    const wrapper = getWrapper();

    expect(wrapper).toMatchSnapshot();
  });

  it('should return html translation in a div', () => {
    const wrapper = getWrapper({ element: 'div' });

    expect(wrapper).toMatchSnapshot();
  });
});
