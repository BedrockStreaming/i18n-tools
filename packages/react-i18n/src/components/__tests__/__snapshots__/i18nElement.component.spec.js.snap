// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n.renderProps should return html translation 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div>foo.bar</div>",
    }
  }
/>
`;

exports[`i18n.renderProps should return html translation in a div 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div>foo.bar</div>",
    }
  }
/>
`;
